src/emit/soroban/mod.rs

```diff
@@ -63,7 +63,7 @@ impl SorobanTarget {
         );
         binary.internalize(export_list.as_slice());
 
-        Self::emit_initializer(&mut binary, ns);
+        Self::emit_initializer(&mut binary, ns, contract.constructors(&ns).first());
 
```

Comment:
Another approach I have in mind is to get the constructor `ControlFlowGraph` itself and prepend it with a call to `storage_initializer`, but this won't work either because the contract is immutable.

We could clone it and re-export it, but I’m not sure if that’s the best approach.